### PR TITLE
Fix #714

### LINKED ISSUE #714: port() not working after initialization if port(0) is provided
This is the failing case:

```
...
port(0);
<routing setup>
....
awaitInitialization();
port() --> is returning 0, although jetty has been correctly initialised with another free port
```

### PR DESCRIPTION
**Pull request by Mirosta**
Calling port() after setting port(0) now returns the correct port
Joining the embedded server can be done with the join method

### CODE REVIEW

File: src/main/java/spark/Service.java [hunk 1]
```diff
@@ -453,10 +453,15 @@ public synchronized void init() {
...
                             sslStores,
-                            latch,
                             maxThreads,
                             minThreads,
                             threadIdleTimeoutMillis);
+                    try {
+                        latch.countDown();
+                        server.join();
+                    } catch (InterruptedException e) {
+                        LOG.error("server interrupted", e);
```
Comment (jakaarl) [inline src/main/java/spark/Service.java]: Shouldn't the interrupt flag be restored here?

Comment (Mirosta) [inline src/main/java/spark/Service.java]: Good point, fixed

**Comment by perwendel** [comment]
:+1: 
Thanks!

**Comment by Mirosta** [comment]
No problem

**Pull request merged**